src/lex.rs

```diff
@@ -20,7 +20,7 @@ impl Operator {
     pub fn operate(self, x: f64, y: f64) -> Result<f64, CalcError> {
         if self.token == '/' && y == 0. {
             return Err(CalcError::Math(Math::DivideByZero));
-        } else if (self.token == '!' && x < 0.0) || (self.token == '!' && x.fract() != 0.0) {
+        } else if self.token == '!' && (x.fract() != 0.0 || x < 0.0) {
```

Comment:
```suggestion
        } else if self.token == '!' && (x < 0.0 || x.fract() != 0.0) {
```
I think chances of x negative is higher than fract not equal to zero.